nunaliit2-couch-onUpload/src/main/java/ca/carleton/gcrc/couch/onUpload/parser/ContentTypeDetector.java

```diff
@@ -9,132 +10,150 @@
-		try (FileInputStream fis = new FileInputStream(file);
-			 InputStreamReader reader = new InputStreamReader(fis, StandardCharsets.UTF_8)) {
-			GeoJsonParser parser = new GeoJsonParser();
-			try {
-				parser.parse(reader);
-				isGeoJson = true;
-			}
-			catch (Exception e) {
-				// Ignore - not a geo json file.
-			}
-		}
-		catch (FileNotFoundException e) {
-			log.error("File not found", e);
-		}
-		catch (IOException e) {
-			log.error("Error reading file", e);
-		}
-
-		return isGeoJson;
-	}
+public class ContentTypeDetector {
+    private static final Logger log = LoggerFactory.getLogger(ContentTypeDetector.class);
+
+    private ContentTypeDetector() {
+    }
+
+    /**
+     * Nunaliit code depends on specific "file class" instead of mime types. This
+     * method returns the ones we support by
+     * examining the file. Possible Nunaliit "file classes" are image, video, audio,
+     * pdf, geojson, gpx.
+     *
+     * @param file File to determine Nunaliit file class.
+     * @return The Nunaliit custom file class.
+     */
+    public static String detectNunaliitFileClass(File file) {
+        String fileClass = "";
+        MediaType mediaType = detectMimeType(file);
+
+        // image, video, audio
+        if (mediaType != null) {
+            if (!mediaType.getType().equalsIgnoreCase("application")) {
+                fileClass = mediaType.getType();
+            } else if (mediaType.getSubtype().equalsIgnoreCase("xml") && isGpx(file)) {
+                fileClass = "gpx";
+            } else if (mediaType.getSubtype().equalsIgnoreCase("json") && isGeoJson(file)) {
+                fileClass = "geojson";
+            } else if (mediaType.getSubtype().equalsIgnoreCase("pdf")) {
+                fileClass = "pdf";
+            }
+        }
+
+        return fileClass;
+    }
+
+    /**
+     * Determine the mime type of given file.
+     *
+     * @param file The file to detect mime type of.
+     * @return The mime type of the file, or null if it was undetermined.
+     */
+    public static MediaType detectMimeType(File file) {
+        if(file == null) {
+            log.warn("File is null, cannot determine mime type");
+            return null;
+        }
+
+        MediaType mediaType = null;
+        try (TikaInputStream tikaInputStream = TikaInputStream.get(file.toPath())) {
+            TikaConfig tika = new TikaConfig();
+            Metadata metadata = new Metadata();
+            metadata.set(Metadata.RESOURCE_NAME_KEY, file.getName());
+            mediaType = tika.getDetector().detect(tikaInputStream, metadata);
+
+            MultimediaClass aClass = MimeUtils.getMultimediaClassFromMimeType(mediaType.toString());
+            if (MultimediaClass.AUDIO == aClass || MultimediaClass.VIDEO == aClass) {
+                List<FileStream> fileStreams = FFprobeProcessor.getFileStreams(file);
+                if(fileStreams == null) {
+                    System.out.println("FILE STREAMS EMPTY");
```

Comment:
Change to use logger

Reply:
The message to be logged could also be a bit more informative (or not in all uppercase at the very least)